DynamicMemory/Exercise5/Student.cpp

```diff
@@ -0,0 +1,43 @@
+#include "Student.h"
+#include <string.h>
+
+Student::Student() = default;
+
+Student::~Student()
+{
+	memcpy_s(StudentName, strlen(StudentName) + 1, "", strlen(StudentName) + 1);
+	StudentAge = 0;
+}
+
+std::string Student::GetName() const
+{
+	return StudentName;
+}
+
+int Student::GetAge() const
+{
+	return StudentAge;
+}
+
+void Student::SetName(const char* InName)
+{
+
+	size_t Length = strlen(InName);
+	if (NameAddress != nullptr)
+	{
+		//Exercise 9 Solution
+		// I stored the old location and clean it
+		memcpy_s(NameAddress, strlen(NameAddress) + 1, "", strlen(NameAddress) + 1);
```

Comment:
Hummmm I think this is not the way to solve the leak, maybe you have to solve it with a destructor

Reply:
yeap later, I realized that this way did not work, and also created some errors 🤡 I Fixed it